File: kotlinamesj/src/main/java/com/nagopy/android/kotlinames/property/nullable/KNullableIntegerPropertyName.java
package com.nagopy.android.kotlinames.property.nullable;

import com.nagopy.android.kotlinames.KNullablePropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KNullableIntegerPropertyName implements KNullablePropertyName<Integer>, KSortablePropertyName<Integer> {

    private final String name;

    public KNullableIntegerPropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/nullable/KNullableIntegerPropertyName.kt
package com.nagopy.android.kotlinames.property.nullable

import com.nagopy.android.kotlinames.KNullablePropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KNullableIntegerPropertyName(val name: String) : KNullablePropertyName<Int>, KSortablePropertyName<Int> {

    override fun name(): String = name

}
